docs/high-availability/provision.sh

```diff
@@ -0,0 +1,181 @@
+    ( journalctl -u kubelet | grep -A 20 -B 20 Fail || echo "no failure in logs")
+    echo "----------- END DEBUG OF KUBELET ----------------------------"
+
+    ( curl http://kube0.ha:2379 > /tmp/curl_output || echo "failed etcd!!!" )
+    if [ -s /tmp/curl_output ]; then
+         echo "etcd success"
+    else
+         echo "etcd failure.  exit!"
+         exit 100
+    fi
+
+}
+
+function k8petstore {
+     ### run K8petstore .  Should work perfectly IFF flannel and so on is setup properly.
+    wget https://raw.githubusercontent.com/GoogleCloudPlatform/kubernetes/release-0.17/examples/k8petstore/k8petstore.sh
+     chmod 777 k8petstore.sh
+    ./k8petstore.sh
+}
+
+function write_api_server_config {
+    touch /var/log/kube-apiserver.log
+    mkdir -p -m 777 /srv/kubernetes/
+
+    ### We will move files back and forth between the /srv/kube.. directory.
+    ### That is how we modulate leader.  Each node will continously either
+    ### ensure that the manifests are in this dir, or else, are in the kubelet manifest dir.
+    cp /vagrant/kube-scheduler.manifest /vagrant/kube-controller-manager.manifest /srv/kubernetes
+
+    ### All nodes will run an API Server.  This is because API Server is stateless, so its not a problem
+    ### To serve it up everywhere.
+    cp /vagrant/kube-apiserver.manifest /etc/kubernetes/manifests/
+}
+
+function write_podmaster_config {
+    touch /var/log/kube-scheduler.log
+    touch /var/log/kube-controller-manager.log
+
+    ### These DO NOT go in manifest. Instead, we mount them here.
+    ### We let podmaster swap these in and out of the manifests directory
+    ### based on its own internal HA logic.
+    cp /vagrant/kube-controller-manager.manifest /srv/kubernetes/
+    cp /vagrant/kube-scheduler.manifest /srv/kubernetes/
+
+    #### Finally, the podmaster is the mechanism for election
+    cp /vagrant/podmaster.json /etc/kubernetes/manifests/
+}
+
+function poll {
+    ### wait 10 minutes for kube-apiserver to come online
+    for i in `seq 1 600`
+    do
+           sleep 2
+            echo $i
+           ### Just testing that the front end comes up.  Not sure how to test total entries etc... (yet)
+           ( curl "localhost:8080" > result || echo "failed on attempt $i, retrying again.. api not up yet. " )
+           ( cat result || echo "no result" )
+           if ( cat result | grep -q api ) ; then
+                break
+           else
+              echo "continue.."
+           fi
+   done
+   if [ $i == 600 ]; then
+        exit 2
+   fi
+}
+
+function install_components {
+    ### etcd node - this node only runs etcd in a kubelet, no flannel.
+    ### we dont want circular dependency of docker -> flannel -> etcd -> docker
+    if [ "`hostname`" == "kube0.ha" ]; then
+            write_etcd_manifest
+            start_kubelet
+
+            ### precaution to make sure etcd is writable, flush iptables.
+             iptables -F
```

Comment:
I'm getting almost every time I run this now:
```
==> kube2: + iptables -F
==> kube2: Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
If ti's a precation we should `|| true` it or just remove it.